packages/data/src/components/use-select/index.js

```diff
@@ -114,6 +114,9 @@ export default function useSelect( _mapSelect, deps ) {
 			errorMessage += 'Original stack trace:';
 
 			throw new Error( errorMessage );
+		} else {
+			// eslint-disable-next-line no-console
+			console.error( errorMessage );
```

Comment:
Instead of logging the error could we do `throw new Error( errorMessage );`? If yes we can simply remove the else and move `throw new Error( errorMessage );` outside the if.

Reply:
No, that would remove the protection against zombie child updates.